File: src/market-tracker-android/app/src/main/java/pt/isel/markettracker/domain/model/market/price/PriceAlert.kt
package pt.isel.markettracker.domain.model.market.price

import java.time.LocalDateTime

data class PriceAlert(
    val id: String,
    val product: ProductAlert,
    val store: StoreItem,
    val priceThreshold: Int,
    val createdAt: LocalDateTime,
)

data class ProductAlert(
    val id: String,
    val name: String,
    val imageUrl: String,
)

data class PriceAlertId(
    val value: String,
)

data class StoreItem(
    val id: Int,
    val name: String,
)

File: src/market-tracker-webapi/Application/Domain/Schemas/Market/Retail/Sales/Pricing/PriceAlert.cs
namespace market_tracker_webapi.Application.Domain.Schemas.Market.Retail.Sales.Pricing;

public record PriceAlert(
    PriceAlertId Id,
    Guid ClientId,
    PriceAlertProduct Product,
    PriceAlertStore Store,
    int PriceThreshold,
    DateTime CreatedAt
);

public record PriceAlertProduct(
    string Id,
    string Name,
    string ImageUrl
);

public record PriceAlertStore(
    int Id,
    string Name
);

public record PriceAlertId(
    string Value
);